docs: enhanced network failover documentation (#4391)
eclipse/kura: docs/gateway-configuration/network-failover.md

192.168.2.0     0.0.0.0         255.255.255.0   U     100    0        0 eth0
```

The `metric` flag will correspond to the set **WAN Priority**. *NetworkManager* will always prioritize lower metric routes.

> ## Work Modalities

Reviewer: This sounds weird in english, I would go for:

```suggestion
## Operating modes
```

The *NetworkManager* failover mechanism can work at two different levels:

> - by detecting disruptions at physical level;

Reviewer: ```suggestion
- by detecting disruptions at physical level (i.e. an ethernet cable that disconnects);
```

*NetworkManager* brings a network interface down when it detects the loss of its physical link. In such case, the next highest priority interface is selected as the main one.

When the connectivity check fails, *NetworkManager* penalizes the metric of that interface in the routing table. *NetworkManager* continues to perform connectivity checks over all the other interfaces. As soon as the connectivity is restored over a previously failing interface, the metric is also restored to the original value and the routing table goes back to the original state.

### Configuring the connectivity check

The **connectivity check** is enabled by default in Kura and is configured to probe the connection to `http://network-test.debian.org/nm` every 60 seconds. To set a specific URI and a different interval edit `/etc/NetworkManager/conf.d/99kura-nm.conf` (reference [NetworkManager](https://www.digi.com/resources/documentation/digidocs/90001548/reference/yocto/r_network_failover.htm?TocPath=Digi%20Embedded%20Yocto%7CSystem%20development%7CSoftware%20extensions%7C_____3)):

```
[connectivity]
uri=http://network-test.debian.org/nm
interval=60
response="NetworkManager is online"
```

> The **interval** minimun is 60 seconds, if missing it defaults to 300 seconds.

Reviewer: ```suggestion
The minimum **interval** is 60 seconds, if no interval is specified it defaults to 300 seconds.
```

> The **response** should match what the **uri** is returning when probed. Some examples of web pages with *NetworkManager* responses:

Reviewer: Be consistent with capitalization and formatting

```suggestion
The **response** should match what the URI is returning when probed. Some examples of web pages with *NetworkManager* responses:
```
